### PR TITLE
Support running a local instance with Docker and Docker Compose

### PR DESCRIPTION
**Pull request by harlev**
Added the following
* `Dockerfile` to build an image for the Jsonbox application.
* `docker-compose.yml` to run both Jsonbox, and its Mongo DB dependency.
* `wait-for-it.sh` from [here](https://github.com/vishnubob/wait-for-it) recommended by the [Docker Documentation](https://docs.docker.com/compose/startup-order/) as a way to wait for a service to be ready. In our case we need it to wait for Mongo DB to be responsive.
* Updated `README.md` with instructions for usage.

In short, just call `docker-compose up` if you have all the docker prerequisites.
The server will be available on http://localhost:3000
If needed, you could change the port in the `docker-compose.yml` file

### CODE REVIEW

**Comment by vasanthv** [comment]
@harlev having the Dockerfile in the main folder might break the current deployment. Can we have them in a separate folder with a separate Readme?

**Comment by harlev** [comment]
moved files to `docker` directory and created a separate readme file.  
Tested it all works in the new location

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM node:latest
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+COPY package.json /usr/src/app/
+RUN npm install
+COPY . /usr/src/app
+EXPOSE 3000
+CMD [ "./utils/wait-for-it.sh", "mongo:27017", "--", "npm", "start" ]
```
Comment (vasanthv) [inline docker/Dockerfile]: `utils` folder needs to be moved here, as it's looking for the relative path.

Comment (harlev) [reply docker/Dockerfile]: The `CMD` here is running relative to the root of the repository.  
`COPY` is copying everything from the root, then `./utils` is referenced from there
